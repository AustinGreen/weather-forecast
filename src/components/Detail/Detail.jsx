import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import TransparentBackground from '../../shared/Backgrounds';
import { Link } from 'react-router-dom';

const BackButton = styled.div`
  position: absolute;
  top: 10px;
  left: 10px;
`;

const Detail = ({ location }) => {
  const { forecastResult } = location.state;
  const dayIndex = location.hash.replace('#', '');
  const detailedForecast = forecastResult.list[dayIndex];
  const forecastPath = location.pathname.replace(/\/details/, '');
  return (
    <div>
      <BackButton className="button is-black">
        <Link to={forecastPath}>
          &lt; Back
        </Link>
      </BackButton>
      <TransparentBackground className="section column is-half is-offset-one-quarter is-large has-text-centered">
        <div className="message">
          <p className="message-body title">
            The high is {Math.round(detailedForecast.temp.max)}°F, and the low is
            {' '}{Math.round(detailedForecast.temp.min)}°F.
          </p>
        </div>
      </TransparentBackground>
    </div>
  );
};

Detail.propTypes = {
  location: PropTypes.shape({
    hash: PropTypes.string,
    state: PropTypes.shape({
      forecastResult: PropTypes.object,
    }),
  }).isRequired,
};

export default Detail;
