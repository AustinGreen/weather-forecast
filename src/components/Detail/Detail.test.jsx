/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Detail from './Detail';
import { shallow } from 'enzyme';
import forecastTestData from './fixtures';

describe('Detail', () => {
  it('renders', () => {
    const div = document.createElement('div');
    const forecastResult = forecastTestData;
    ReactDOM.render(
      <Detail
        location={{
          hash: '#1',
          state: { forecastResult },
        }}
      />,
      div,
    );
  });

  test('snapshots', () => {
    const forecastResult = forecastTestData;
    const component = renderer.create(
      <Detail
        location={{
          hash: '#1',
          state: { forecastResult },
        }}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
