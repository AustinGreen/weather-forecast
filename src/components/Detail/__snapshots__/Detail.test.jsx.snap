exports[`Detail snapshots 1`] = `
<div>
  <h1>
    Detail view for
  </h1>
  <ul>
    <li>
      dt: 1492599600
    </li>
    <li>
      temp: [object Object]
    </li>
    <li>
      pressure: 1044.05
    </li>
    <li>
      humidity: 57
    </li>
    <li>
      weather: [object Object]
    </li>
    <li>
      speed: 1.86
    </li>
    <li>
      deg: 88
    </li>
    <li>
      clouds: 8
    </li>
  </ul>
</div>
`;
